tests/garbage_collect/test_garbage_collect.py

```diff
@@ -227,11 +230,74 @@ def test_arc_garbage_collect(mock_logger):
     with patch("sys.argv", arg):
         with _patch_pulp_client(controller.client):
             gc.main()
+
     updated_rpm = list(client.get_repository("all-rpm-content").search_content())
     assert len(updated_rpm) == 1
     mock_logger.info.assert_any_call("Old all-rpm-content deleted: %s", rpm2.name)
 
 
+def test_arc_garbage_collect_in_batches(mock_logger):
+    """deletes relevant all-rpm-content content in batches"""
+    gc_module.UNASSOCIATE_BATCH_LIMIT = 5
```

Comment:
Nothing will automatically undo this at the end of the test, so if any other tests are influenced by UNASSOCIATE_BATCH_LIMIT, their behavior will differ depending on whether the current process has run this test first.  Even if it doesn't affect any existing test, it's a trap for developers adding more tests later.

It's best to avoid that and have tests undo whatever changes they make.  You can use monkeypatch fixture to do this easily, something like:

```python
def test_arc_garbage_collect_in_batches(mock_logger, monkeypatch: pytest.MonkeyPatch):
    """deletes relevant all-rpm-content content in batches"""
    monkeypatch.setattr(gc_module, 'UNASSOCIATE_BATCH_LIMIT', 5)
```

Then pytest will take care of undoing this for you at the end of the test.